MyPy: Fix missing return type annotations in API endpoints

## Problem

13 MyPy errors due to missing return type annotations across API endpoints and main application file.

## Files Affected

- `app/api/v1/endpoints/organizations.py` - 5 errors
- `app/api/v1/endpoints/health.py` - 1 error  
- `app/main.py` - 7 errors

## Errors

### organizations.py
```
app/api/v1/endpoints/organizations.py:47: error: Function is missing a return type annotation  [no-untyped-def]
app/api/v1/endpoints/organizations.py:98: error: Function is missing a return type annotation  [no-untyped-def]
app/api/v1/endpoints/organizations.py:133: error: Function is missing a return type annotation  [no-untyped-def]
app/api/v1/endpoints/organizations.py:201: error: Function is missing a return type annotation  [no-untyped-def]
app/api/v1/endpoints/organizations.py:289: error: Function is missing a return type annotation  [no-untyped-def]
```

### health.py
```
app/api/v1/endpoints/health.py:16: error: Function is missing a return type annotation  [no-untyped-def]
```

### main.py
```
app/main.py:27: error: Function is missing a return type annotation  [no-untyped-def]
app/main.py:125: error: Function is missing a return type annotation  [no-untyped-def]
app/main.py:139: error: Function is missing a return type annotation  [no-untyped-def]
app/main.py:157: error: Function is missing a return type annotation  [no-untyped-def]
app/main.py:157: error: Function is missing a type annotation for one or more arguments  [no-untyped-def]
app/main.py:201: error: Function is missing a type annotation  [no-untyped-def]
```

## Solution

Add return type annotations to all functions. Most should return FastAPI response types or `None`.

**Example fix:**
```python
# Before
def health_check():
    return {"status": "healthy"}

# After
def health_check() -> dict[str, str]:
    return {"status": "healthy"}
```

## Difficulty

**Easy** - Straightforward type annotations, no complex logic changes.

## Acceptance Criteria

- [ ] All 13 functions have proper return type annotations
- [ ] MyPy passes with 0 errors in these 3 files
- [ ] All existing tests pass unchanged

## Related

Part of #139 - Fix Backend MyPy type checking errors (76 remaining after #142)

--- a/apps/api/app/api/v1/endpoints/health.py
+++ b/apps/api/app/api/v1/endpoints/health.py
@@ -13,7 +13,7 @@ router = APIRouter()
 
 
 @router.get("/health", tags=["Health"])
-async def health_check_detailed(db: Session = Depends(get_db)):
+async def health_check_detailed(db: Session = Depends(get_db)) -> dict[str, str]:
     """
     Detailed health check with database connectivity test.
 

--- a/apps/api/app/api/v1/endpoints/organizations.py
+++ b/apps/api/app/api/v1/endpoints/organizations.py
@@ -49,7 +49,7 @@ def create_organization(
     current_user: AdminOnly,
     request: Request,
     db: Session = Depends(get_db),
-):
+) -> OrganizationResponse:
     """
     Create a new organization.
 
@@ -98,7 +98,7 @@ def create_organization(
 def list_organizations(
     current_user: AuthenticatedUser,
     db: Session = Depends(get_db),
-):
+) -> List[OrganizationResponse]:
     """
     List organizations.
 
@@ -135,7 +135,7 @@ def get_organization(
     current_user: AuthenticatedUser,
     request: Request,
     db: Session = Depends(get_db),
-):
+) -> OrganizationResponse:
     """
     Get a single organization by ID.
 
@@ -204,7 +204,7 @@ def update_organization(
     current_user: AdminOnly,
     request: Request,
     db: Session = Depends(get_db),
-):
+) -> OrganizationResponse:
     """
     Update an organization by ID.
 
@@ -291,7 +291,7 @@ def delete_organization(
     current_user: AdminOnly,
     request: Request,
     db: Session = Depends(get_db),
-):
+) -> None:
     """
     Delete an organization by ID.
 

--- a/apps/api/app/main.py
+++ b/apps/api/app/main.py
@@ -2,11 +2,12 @@
 FastAPI main application entry point.
 """
 
+from collections.abc import AsyncGenerator
 from contextlib import asynccontextmanager
 from datetime import datetime, timezone
 from uuid import uuid4
 
-from fastapi import FastAPI, HTTPException
+from fastapi import FastAPI, HTTPException, Request
 from fastapi.middleware.cors import CORSMiddleware
 from fastapi.responses import JSONResponse
 import logging
@@ -24,7 +25,7 @@ logger = logging.getLogger(__name__)
 
 
 @asynccontextmanager
-async def lifespan(app: FastAPI):
+async def lifespan(app: FastAPI) -> AsyncGenerator[None, None]:
     """
     FastAPI lifespan context manager for startup and shutdown logic.
 
@@ -122,7 +123,7 @@ app.add_middleware(
 
 # Root endpoint
 @app.get("/", tags=["Root"])
-async def root():
+async def root() -> dict[str, str]:
     """Root endpoint - API information."""
     return {
         "name": settings.PROJECT_NAME,
@@ -136,7 +137,7 @@ async def root():
 
 # Health check endpoint
 @app.get("/health", tags=["Health"])
-async def health_check():
+async def health_check() -> dict[str, str]:
     """
     Health check endpoint for monitoring and load balancers.
 
@@ -154,7 +155,7 @@ async def health_check():
 
 # Custom HTTPException handler to unwrap detail field
 @app.exception_handler(HTTPException)
-async def http_exception_handler(request, exc: HTTPException):
+async def http_exception_handler(request: Request, exc: HTTPException) -> JSONResponse:
     """
     Custom HTTPException handler to unwrap the detail field.
 
@@ -198,7 +199,7 @@ async def http_exception_handler(request, exc: HTTPException):
 
 # Global exception handler
 @app.exception_handler(Exception)
-async def global_exception_handler(request, exc):
+async def global_exception_handler(request: Request, exc: Exception) -> JSONResponse:
     """
     Global exception handler for unhandled errors.
 

